lib/autoproj/autobuild_extensions/git.rb

```diff
@@ -87,7 +87,7 @@ def snapshot_against_remote(package, options = Hash.new)
                 if options[:exact_state] && !info['tag']
                     info['commit'] = rev_parse(package, 'HEAD')
                 end
-                info
+                info.delete_if { |_, v| v.nil? }
```

Comment:
The `nil`'s here as important. Without them, the override would not remove e.g. a tag that would have been set in a package set.

Reply:
Doesn't the `commit` field do that?

Reply:
> Doesn't the commit field do that?

But, then, how do you remove the commit to favor a tag ?

Reply:
Not sure I understand. The generated version file will contain `commit` or `tag` (never both). So, if autoproj really favors what is set in the override file, that should remove whatever is set in the package_set. Am I missing something?

Reply:
> Am I missing something?

Reply:
Perfectly clear. Thanks!

Reply:
For reference: https://doudou.github.io/rock_website/workspace/add_packages.html#version_control_resolution